setup.py

```diff
@@ -183,6 +136,70 @@ def load_dll(libname, fallbacks=None):
         include_package_data=True,
     )
 
+# Build cython extensions, which require development parameters
+include_dirs = [get_config_var('INCLUDEDIR')]
+library_dirs = []
+libraries = []
+extra_link_args = []
+
+# Use geos-config utility to get development parameters
+# (Note: not available for Windows, since it is a shell script)
+geos_config = os.environ.get('GEOS_CONFIG', 'geos-config')
+log.debug('geos_config: %s', geos_config)
+
+
+def get_geos_config(option):
+    try:
+        stdout, stderr = subprocess.Popen(
+            [geos_config, option],
+            stdout=subprocess.PIPE, stderr=subprocess.PIPE).communicate()
+    except OSError as ex:
+        # e.g., [Errno 2] No such file or directory
+        raise OSError(
+            'Could not find geos-config %r: %s' % (geos_config, ex))
+    if stderr and not stdout:
+        raise ValueError(stderr.strip())
+    if sys.version_info[0] >= 3:
+        result = stdout.decode('ascii').strip()
+    else:
+        result = stdout.strip()
+    log.debug('%s %s: %s', geos_config, option, result)
+    return result
+
+try:
+    # Get the version from geos-config. Show error if this version tuple is
+    # different to the GEOS version loaded from the dynamic library.
+    geos_config_version_string = get_geos_config('--version')
+    res = re.findall(r'(\d+)\.(\d+)\.(\d+)', geos_config_version_string)
+    geos_config_version = tuple(int(x) for x in res[0])
+    if geos_config_version != geos_version:
+        log.error("The GEOS dynamic library version is %s %s,",
+                  geos_version_string, geos_version)
+        log.error("but the version reported by %s is %s %s.", geos_config,
+                  geos_config_version_string, geos_config_version)
+        sys.exit(1)
+except OSError as ex:
+    log.error(ex)
+    log.error('Cannot find geos-config to get headers and check version.')
+    log.error('If available, specify a path to geos-config with a '
+              'GEOS_CONFIG environment variable')
+    geos_config = None
+
+
+if geos_config:
```

Comment:
Also, a few lines up `geos_config = None` if the utility does not run / is not found.